pkg/controller/ps/controller.go

```diff
@@ -817,14 +817,6 @@ func (r *PerconaServerMySQLReconciler) reconcileGroupReplication(ctx context.Con
 
 func (r *PerconaServerMySQLReconciler) cleanupOutdatedServices(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL, exposer Exposer) error {
 	log := logf.FromContext(ctx).WithName("cleanupOutdatedServices")
-
-	if !cr.HAProxyEnabled() || cr.Spec.Proxy.HAProxy.Size == 0 {
-		svc := haproxy.Service(cr, nil)
-		if err := r.Client.Delete(ctx, svc); err != nil && !k8serrors.IsNotFound(err) {
-			return errors.Wrapf(err, "delete HAProxy svc %s", svc.Name)
-		}
-	}
```

Comment:
removing this is a good idea, idk why we put a special case for haproxy in this function like this

Reply:
I think this was here because we don't have `haproxy.Exposer(*cr)`

Reply:
Thanks @pooknull for the help)